man/systemd.service.xml

```diff
@@ -897,6 +897,28 @@
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>RestartMode=</varname></term>
+
+        <listitem>
+          <para>Takes a string value that specifies how a failing service should restart:
+            <itemizedlist>
+              <listitem><para>If set to <option>normal</option> (the default), the service restarts by
+              going through a failed state.</para></listitem>
+
+              <listitem><para>If set to <option>quick</option>, the service restarts without
+              going through a failed state.
+              <varname>ExecStopPost=</varname> and <varname>OnSuccess=</varname> are invoked.
```

Comment:
I wonder if we want to invoke `OnSuccess=` in  this scenario.

Reply:
I'd say we shouldn't. And `OnSuccess=` is actually getting triggered as a side-effect of #27722. If that's fixed, with this patch `OnSuccess=` will be skipped too.